cookbooks/cdo-nodejs/recipes/default.rb

```diff
@@ -3,12 +3,53 @@
 # Recipe:: default
 #
 
-node.default['nodejs']['repo'] = "https://deb.nodesource.com/node_#{node['cdo-nodejs']['version']}"
-include_recipe 'nodejs'
+is_ubuntu_18_04 = node['platform'] == 'ubuntu' && node['platform_version'] == '18.04'
+is_node_18 = node['cdo-nodejs']['version'] == '18.x'
 
-# Keep nodejs up to date
-package 'nodejs' do
-  action :upgrade
+if is_ubuntu_18_04 && is_node_18
+  # No nodesource binary for NodeJS 18 on Ubuntu 18.04
+  # Compile from source instead, using gcc-8
+
+  build_essential 'install build tools'
+
+  package %w(libssl-dev python g++ g++-8 gcc-8)
+
+  ark 'nodejs-source' do
+    url 'https://nodejs.org/dist/v18.16.0/node-v18.16.0.tar.gz'
+    checksum '6a4f5c5d76e5c50cef673099e56f19bc3266ae363f56ca0ab77dd2f3c5088c6d'
+    version '18.16.0'
+    make_opts ['-j 8']
```

Comment:
tiny nit: any way this value can be determined dynamically rather than hardcoded? My understanding is that it should ideally be equal to the number of processors available to us